### ISSUE TITLE
feature request: min supported D version

### ISSUE THREAD

**Issue by SingingBush** [dlang/dub#1097]
It would be handy if a project that is `targetType "library"` would have to state in it's dub.json/sdl which D version is required. So if my project (or any dependency in my project) needs dmd 2.072 or equivelant and I attempt to compile with an older version such as dmd 2.071 or LDC 1.1.0, dub would ouput a message to let me know exactly what the problem is and which D version is needed.

so something like:

```json
{
	"name": "somelib",
	"targetType": "library",
	"description": "a lib that uses latest D features",
	"dlang": "2.073"
}
```

Then if somelib is in the dependency tree dub knows it needs 2.073 or greater and makes sure the compiler is suitable before continuing. Where there are multiple dependencies that all have different versions for a minimum compiler, dub would the highest number found.

**Comment by andre2007** [comment]
Will come with PR https://github.com/dlang/dub/pull/1571

**Comment by andre2007** [comment]
Implemented, see pr #1571
